code/src/hooks/use-sign-up/useSignUp.ts

```diff
@@ -0,0 +1,30 @@
+import { useSignUpMutation } from "@/store/api/authApi";
+import { SignUpCredentials } from "@/types/auth.types";
+import useSnackbar from "@/hooks/use-snackbar/useSnackbar";
+import { authenticate } from "@/store/slices/userSlice";
+import { useAppDispatch } from "@/hooks/use-redux/useRedux";
+
+const useSignUp = () => {
+  const [signUp, options] = useSignUpMutation();
+  const { openSnackbarWithTimeout } = useSnackbar();
+  const dispatch = useAppDispatch();
+
+  const handleSignUp = async (credentials: SignUpCredentials) => {
+    const { data: userDetails, error } = await signUp(credentials);
```

Comment:
also here. What's happening while it's loading ?

Reply:
You can use `isLoading` externally as you want

Reply:
> You can use `isLoading` externally as you want. It is just and example in PR description

I saw example. But there nothing here in `useSignUp` hook about loading how you get the value for it ?

Reply:
The value for `isLoading` is provided by rtk query. I hope I got your question correctly

Reply:
> The value for `isLoading` is provided by rtk query. I hope I got your question correctly

Interesting approach to get isLoading via destructuring with signUp value and it's everything in array. Just confusing for me I mean `const [signUp, { isLoading: isSigningUp }] = useSignUp()`